integration/tests/test_upgrade.py

```diff
@@ -105,9 +105,29 @@ def success_predicate(topics_available):
     spin(fn, success_predicate)
 
 def write_messages():
+	# kafka may not be ready to accept all msgs, try till all are done
+    def fn(num):
+        try:
+            offset_info = get_kafka_command('topic offsets {}'.format(TOPIC_NAME))
+            offset = int(offset_info[0]['0'])
+            if offset < num:
+               get_kafka_command(
+                   'topic producer_test {} {}'.format(TOPIC_NAME, num - offset))
+            print('producer_test {} msg is created'.format(offset))
+            assert (num - offset) >= 0
+            return num - offset 
+        except RuntimeError:
+            return num
+
+    def success_predicate(left_offset):
+        return (left_offset <= 0, 'producer_test continues....')
+    
     get_kafka_command(
         'topic producer_test {} {}'.format(TOPIC_NAME, NUM_TEST_MSGS)
     )
```

Comment:
first you gotta create the topic with producer_test. you cant call offsets before creating the topic. it will always fail, isnt it? so L108 is not reduntant